### PR TITLE
add TimestampResultCode to mc_api

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

Fog needs to communicate `TimestampResultCode` inside some of its proto objects. Currently the proto equivalent of `TimestampResultCode` lives in internal/fog, but the correct place for it is in this repository, next to all the other proto APIs bundled in `mc_api::external`.

### In this PR
* Add `TimestampResultCode` to `external.proto`
* Add appropriate `From`/`TryFrom` conversions. 

### Future Work
* Use this in internal fog code.

### CODE REVIEW

**Review by garbageslam — APPROVED** [review]

File: api/proto/external.proto [hunk 1]
```diff
@@ -267,3 +267,28 @@ message Receipt {
     // Note: This value is self-reported by the sender and is unverifiable.
     Amount amount = 4;
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// `watcher/api` crate
```
Comment (kylefleming) [inline api/proto/external.proto:272]: Is it worth putting this in a separate `api/proto/watcher.proto` file?

Comment (eranrund) [reply api/proto/external.proto:272]: I don't have a strong preference. Does anyone else does? I believe the idea of `external.proto` is to collect all the proto stuff we want to expose to non-Rust clients.

Comment (kylefleming) [reply api/proto/external.proto:272]: The distinction I was making in my head is that the watcher isn't part of the "core" mobilecoin, similar to how `PrintableWrapper` is in it's own proto file. I'm not sure if there's a benefit to having *all* external types in a single proto file or not? Maybe `external.proto` should really be called something like `core.proto`?

Comment (eranrund) [reply api/proto/external.proto:272]: Okay, I'll move this into `watcher.proto`. Everything else is definitely out of scope for this PR.

**Review by kylefleming — APPROVED** [review]

**Comment by kylefleming** [comment]
Thank you!

**Pull request merged**